sdk/keyvault/keyvault/src/az_keyvault_client.c

```diff
@@ -252,14 +243,16 @@ AZ_NODISCARD az_result az_keyvault_keys_key_get(
 
   uint8_t headers_buffer[4 * sizeof(az_pair)];
   az_span request_headers_span = AZ_SPAN_FROM_BUFFER(headers_buffer);
-
-  // check if url needs to be reset to initial state
-  AZ_RETURN_IF_FAILED(_az_reset_url_to_initial_state(client));
+  // Url buffer
+  uint8_t url_buffer[1024];
```

Comment:
Constant

Reply:
yes

Reply:
that was possible when we had az_span and az_mut_span...
az_span has an `uint8_t *` , so we got an error of `discards-qualifiers` if trying to do
```c
uint8_u const buffer[10];
az_span a = { ._internal = { .ptr = buffer  } }
```
error: initializing 'uint8_t *' (aka 'unsigned char *') with an expression
      of type 'const uint8_t [1024]' discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
  az_span const url_buffer_span = AZ_SPAN_FROM_BUFFER(url_buffer);

And also, this is the buffer where we are mutating url, so it can't be constant